src/Utilities/OperatingSystemFilterHelpers.js

```diff
@@ -1,31 +1,44 @@
+import { mapValues, omit } from 'lodash';
 import { coerce, compare, rcompare } from 'semver';
-import { OS_CHIP } from './index';
+import { OS_CHIP } from './constants';
 
 /* Takes an array of string versions and returns an object in the format
-   required by ConditionalFilter component (group filter) */
-export const toGroupSelectionValue = (osVersions = []) => osVersions.reduce((acc, version) => {
-    const [majorVersion, minorVersion] = version.split('.');
-    acc[`${majorVersion}.0`] = {
-        ...(acc[`${majorVersion}.0`] || {}),
-        [`${majorVersion}.${minorVersion}`]: true
+   required by ConditionalFilter component (group filter);
+   inverse to `getSelectedOsFilterVersions`  */
+export const toGroupSelection = (osVersions = []) => osVersions.reduce((acc, version) => {
+    const [major] = version.split('.');
+    acc[major] = {
+        ...(acc[major] || {}),
+        [version]: true,
+        [major]: Object.values({ ...acc[major] }).filter(v => v !== major).every(Boolean)
```

Comment:
There is some repeated code here that can be found in `src/components/filters/useOperatingSystemFilter`. I think we can create some helper functions that we can use globally and are named well so we know what is being done here. I'm having a hard time looking at this line and knowing what's happening. A well named function could make the difference.

Reply:
yep, true, updated with the last push 👍🏼